client/src/app/model/middleware.js

```diff
@@ -0,0 +1,23 @@
+export const preventDispatchForPerformer = store => next => action => {
+  if (typeof action === 'function') {
+    return next(action);
+  }
+  const { payload } = action;
+
+  if (typeof payload === 'object') {
```

Comment:
I think this if is unnecessary

Reply:
For now, we have to keep it. We have actions that payload isn't an object eg. remove notification.